core/src/main/scala/io/finch/Endpoint.scala

```diff
@@ -56,16 +58,22 @@ import shapeless.ops.hlist.Tupler
  *   Http.server.serve(foobar.toService)
  * }}}
  */
-trait Endpoint[A] { self =>
+abstract class Endpoint[A] extends Serializable { self =>
 
-  type ContentType <: String
+  /**
+   * The [[StateT]] instance underneath this endpoint.
+   */
+  def run: Endpoint.State[A]
```

Comment:
Maybe call this something like `state` or `underlying` to avoid confusion with `.run.run`?